### DISCUSSION TITLE
Lambda Container Error on /var/task/vendor/autoload.php

### DISCUSSION THREAD

**Discussion by wheeleruniverse** [brefphp/bref#824] [Support]
**Description:**
I am trying to deploy my php web application in a serverless architecture using AWS Lambda. I am trying to utilize the new feature for containerized support on Lambda as opposed to the previous method of deploying Bref as a Lambda Layer. 

I followed the instructions in this Blog post: https://mnapoli.fr/aws-lambda-php-docker-containers/, but when I deployed my container on Lambda I am getting the following error on invocation. 

     PHP Fatal error:  require(): Failed opening required '/var/task/vendor/autoload.php' 
     (include_path='.:/opt/bref/lib/php') in /opt/bootstrap on line 17

Running `docker run` on my local laptop is also throwing the same error. 

Here is my repository so you can see I am including everything inside this php directory in my docker container.
https://github.com/wheelers-websites/CloudGuruChallenge_1020/tree/master/php

**How to reproduce:**

* AWS Service: Containerized Lambda
* Bref Version: bref/php-74-fpm

1. Create Docker Image

```Docker
FROM bref/php-74-fpm
COPY . /var/task
CMD _HANDLER=index.php /opt/bootstrap
```
2. Run Docker Image

```Docker
docker run
```

![Bref_Error_20201214](https://user-images.githubusercontent.com/71364086/102167002-7624c780-3e5b-11eb-9f08-8ca2d0fc9069.PNG)
![DockerVersion_20201214](https://user-images.githubusercontent.com/71364086/102167003-76bd5e00-3e5b-11eb-814a-6780297955d5.PNG)

**Comment by mnapoli** [comment]
Sorry the blog post is incomplete, I assumed something that may not be obvious to everyone 😅 

You need to install bref in your project via:

```
composer require bref/bref
```

first. That will create a `composer.json` file and `vendor/` directory.

Let me know if that works better.

**Reply by wheeleruniverse** [reply]
Thanks! When I updated my Dockerfile to the following it's seemingly working now.

```Docker
FROM bref/php-74-fpm
RUN curl -s https://getcomposer.org/installer | php
RUN php composer.phar require bref/bref
COPY . /var/task
CMD _HANDLER=index.php /opt/bootstrap
```

I am currently seeing below, but I haven't created the API Gateway yet. I will do that next. Thank you!

```JSON
"errorType": "Bref\\Event\\InvalidLambdaEvent",
"errorMessage": "This handler expected to be invoked with a API Gateway or ALB event. Instead, the handler was invoked with invalid event data: Array\n(\n)\n",
```

**Comment by stloc** [comment]
Hi,
it seems that the error also happens when vendor is in another directory ? like https://github.com/Cockpit-HQ/Cockpit

composer.json
 ```
"config": {
        "vendor-dir": "lib/vendor",
        "optimize-autoloader": true,
        "prefer-dist": true,
        "lock": false,
        "ignore-platform-reqs": true
    }
```

serverless.yml
```
plugins:
    - ./lib/vendor/bref/bref
```

cloudwatch log : Fatal error: Uncaught Error: Failed opening required '/var/task/vendor/autoload.php' (include_path='.:/opt/bref/lib/php') in /opt/bootstrap:37

**Reply by mnapoli** [reply]
@stloc check out https://bref.sh/docs/environment/php.html#custom-vendor-path

**Reply by stloc** [reply]
thanks and sorry Matthieu I should have read the documentation better !

**Reply by mnapoli** [reply]
No problem at all!

**Reply by mnapoli** [reply]
That makes me think that we could actually improve the error message in that case! The error message could contain a link to that documentation page, that would be much better 🤔

I won't have time to work on this soon, but if you want to submit a PR feel free.
